server/misc_tracking.py

```
@@ -0,0 +1,222 @@
+import datetime
+import time
+from server.trackable import Trackable, create_trackable_from_row
+from flask import (
+    Blueprint, jsonify, request, session
+)
+
+from server.db import get_db
+
+bp = Blueprint('api', __name__, url_prefix='/api')
+date_format = '%Y-%m-%d'
+
+
+def get_user_by_api_key(request):
+    db = get_db()
+    api_key = request.json.get('api_key')
+    user = db.execute("SELECT * FROM user WHERE api_key=?",
+                      (api_key,)).fetchone()
+    if user is None:
+        return None
```

Should probably throw an error